File: src/main/java/xyz/oribuin/skyblock/Skyblock.java
package xyz.oribuin.skyblock;

import org.bukkit.plugin.java.JavaPlugin;
import xyz.oribuin.skyblock.database.DatabaseConnector;
import xyz.oribuin.skyblock.database.SQLiteConnector;
import xyz.oribuin.skyblock.managers.CommandManager;
import xyz.oribuin.skyblock.managers.ConfigManager;
import xyz.oribuin.skyblock.managers.MessageManager;
import xyz.oribuin.skyblock.managers.WorldManager;

import java.io.File;

public class Skyblock extends JavaPlugin {

    private static Skyblock instance;

    public static Skyblock getInstance() {
        return instance;
    }

    private DatabaseConnector connector;
    private CommandManager commandManager;
    private ConfigManager configManager;
    private MessageManager messageManager;
    private WorldManager worldManager;

    @Override
    public void onEnable() {
        instance = this;
        this.commandManager = new CommandManager(this);
        this.configManager = new ConfigManager(this);
        this.messageManager = new MessageManager(this);
        this.worldManager = new WorldManager(this);

        this.saveDefaultConfig();
        this.reload();

        this.connector = new SQLiteConnector(this);

        createFolder("schematics");
    }

    private void createFolder(String folderName) {
        File file = new File(this.getDataFolder(), folderName);
        if (!file.exists())
            file.mkdir();
    }

    public void reload() {
        this.commandManager.reload();
        this.configManager.reload();
        this.messageManager.reload();
        this.worldManager.reload();
    }

    public CommandManager getCommandManager() {
        return this.commandManager;
    }

    public ConfigManager getConfigManager() {
        return this.configManager;
    }

    public MessageManager getMessageManager() {
        return this.messageManager;
    }

    public WorldManager getWorldManager() {
        return this.worldManager;
    }

    public DatabaseConnector getConnector() {
        return connector;
    }
}

File: src/main/kotlin/xyz/oribuin/skyblock/Skyblock.kt
package xyz.oribuin.skyblock

import org.bukkit.Bukkit
import xyz.oribuin.skyblock.command.CmdIsland
import xyz.oribuin.skyblock.library.FileUtils
import xyz.oribuin.skyblock.library.HexUtils.colorify
import xyz.oribuin.skyblock.library.OriPlugin
import xyz.oribuin.skyblock.listener.BlockListeners
import xyz.oribuin.skyblock.listener.PlayerListeners
import xyz.oribuin.skyblock.manager.*

class Skyblock : OriPlugin() {
    override fun enablePlugin() {
        this.getManager(ConfigManager::class)
        this.getManager(DataManager::class)
        this.getManager(BorderManager::class)
        this.getManager(HookManager::class)
        this.getManager(IslandManager::class)
        this.getManager(MessageManager::class)
        this.getManager(WorldManager::class)

        // Register plugin commands
        this.registerCommands(CmdIsland(this))

        // Register plugin listeners
        this.registerListeners(
                BlockListeners(this),

                PlayerListeners(this)
        )

        this.reload()
        this.saveDefaultConfig()
        createSchematics("desert", "ice", "default", "mesa", "mushroom", "nether", "plains")

        Bukkit.getScheduler().runTaskTimer(this, Runnable {
            Bukkit.getOnlinePlayers().forEach { player ->
                val islandManager = getManager(IslandManager::class)

                player.sendMessage(colorify("<r:0.5>Is on island: ${(islandManager.getIslandOn(player)?: return@forEach).name}"))
            }
        }, 0, 1)
    }

    override fun disablePlugin() {
        // Unused

    }

    private fun createSchematics(vararg schems: String) {
        for (schematic in schems)
            FileUtils.createDirFile(this, "schematics", "$schematic.schematic")
    }
}
